src/controller/CHIPDeviceController.cpp

```diff
@@ -269,13 +292,33 @@ CHIP_ERROR ChipDeviceController::EstablishSecureSession()
 
     mConState = kConnectionState_SecureConnected;
 
+    if (mStorageDelegate != nullptr)
+    {
+        const char * credentials;
+        const char * address;
+
+        PERSISTENT_KEY_OP(peer, kDeviceCredentialsKeyPrefix, key, credentials = mStorageDelegate->GetKeyValue(key));
+        PERSISTENT_KEY_OP(peer, kDeviceAddressKeyPrefix, key, address = mStorageDelegate->GetKeyValue(key));
+
+        SecurePairingSessionSerialized serialized;
+
+        VerifyOrExit(credentials != nullptr, err = CHIP_ERROR_KEY_NOT_FOUND_FROM_PEER);
+        strncpy(Uint8::to_char(serialized.inner), credentials, sizeof(serialized.inner));
```

Comment:
Do we need to worry about cases when the lengths here don't match?  Especially if `credentials` doesn't fit in `serialized.inner`?

Reply:
Currently, it will not match (or fit) only if someone passes made-up (incorrect) credentials. The `Deserialize` would catch that. I can add a check for the length here.